test-data/exceptions.test

```diff
@@ -77,38 +76,42 @@ def f(x: Optional[A]) -> int:
 def f(x):
     x :: union[A, None]
     r0 :: None
-    r1 :: bool
-    r2 :: int
-    r3 :: A
-    r4 :: None
-    r5, r6 :: bool
-    r7, r8, r9 :: int
+    r1 :: object
+    r2 :: bool
+    r3 :: int
+    r4 :: A
+    r5 :: None
+    r6 :: object
+    r7, r8 :: bool
+    r9, r10, r11 :: int
 L0:
     r0 = None
-    r1 = x is r0
-    if r1 goto L1 else goto L2 :: bool
+    r1 = box(None, r0)
```

Comment:
It would be nice to merge `r0 = None;  r1 = box(None, r0)` into a single opcode (though the C compiler will clean this up).